flow/scripts/cts.tcl

```diff
@@ -53,22 +53,21 @@ detailed_placement
 
 estimate_parasitics -placement
 
-puts "Repair setup violations..."
 # process user settings
 set additional_args ""
 if { [info exists ::env(SETUP_SLACK_MARGIN)] && $::env(SETUP_SLACK_MARGIN) > 0.0} {
-  puts "Setup repair with slack margin $::env(SETUP_SLACK_MARGIN)"
-  append additional_args " -slack_margin $::env(SETUP_SLACK_MARGIN)"
+  puts "Setup slack margin $::env(SETUP_SLACK_MARGIN)"
+  append additional_args " -setup_margin $::env(SETUP_SLACK_MARGIN)"
 }
-eval repair_timing -setup $additional_args
-
-puts "Repair hold violations..."
-# process user settings
-set additional_args ""
 if { [info exists ::env(HOLD_SLACK_MARGIN)] && $::env(HOLD_SLACK_MARGIN) > 0.0} {
-  puts "Hold repair with slack margin $::env(HOLD_SLACK_MARGIN)"
-  append additional_args " -slack_margin $::env(HOLD_SLACK_MARGIN)"
+  puts "Hold slack margin $::env(HOLD_SLACK_MARGIN)"
+  append additional_args " -hold_margin $::env(HOLD_SLACK_MARGIN)"
 }
+
+puts "Repair setup violations..."
+repair_timing -setup {*}$additional_args
+
+puts "Repair hold violations..."
 repair_timing -hold {*}$additional_args
```

Comment:
You can just combine these into one call now that you are using the new flags:

repair_timing {*}$additional_args

Reply:
Makes sense, updated.